gdal/src/test/scala/geotrellis/contrib/vlm/gdal/GDALDataPathSpec.scala

```diff
@@ -335,14 +345,22 @@ class GDALDataPathSpec extends FunSpec with Matchers {
 
         GDALDataPath(uri).vsiPath should be (expectedPath)
       }
+    }
+  }
 
-      it("relative path that points to zip with ! uri") {
-        val filePath = "../../test-files/data.zip"
-        val uri = s"$filePath!$fileName"
-        val expectedPath = s"/vsizip/$filePath/$fileName"
+  describe("Formatting VSI paths") {
+    it("should parse a VSI path") {
+      val filePath = "/vsihdfs/hdfs://data/my-data/data.tif"
+      val expectedPath = s"/vsihdfs/hdfs://$filePath"
 
-        GDALDataPath(uri).vsiPath should be (expectedPath)
-      }
+      GDALDataPath(expectedPath).vsiPath should be (expectedPath)
+    }
+
+    it("should parse a chained VSI path") {
+      val filePath = "/vsizip//vsis3/data/my-data/data.zip"
+      val expectedPath = s"/vsizip//vsis3/$filePath"
+
+      GDALDataPath(expectedPath).vsiPath should be (expectedPath)
```

Comment:
These tests look odd, `expectedPath` would be `/vsizip//vsis3//vsizip//vsis3/data/my-data/data.zip` which isn't a thing

Reply:
These are incorrect tests back ported from the previous PR